OMEdit/OMEditLIB/Animation/ViewerWidget.cpp

```diff
@@ -224,247 +224,260 @@ void ViewerWidget::mousePressEvent(QMouseEvent *event)
-void ViewerWidget::pickShape(int x, int y)
+void ViewerWidget::pickVisualizer(int x, int y)
 {
-  //std::cout<<"pickShape "<<x<<" and "<<y<<std::endl;
+  //std::cout<<"pickVisualizer "<<x<<" and "<<y<<std::endl;
   osgUtil::LineSegmentIntersector::Intersections intersections;
-  if (mpSceneView->computeIntersections(mpSceneView->getCamera(),osgUtil::Intersector::WINDOW , x, y, intersections)) {
+  if (mpSceneView->computeIntersections(mpSceneView->getCamera(), osgUtil::Intersector::WINDOW, x, y, intersections)) {
     //take the first intersection with a facette only
-    osgUtil::LineSegmentIntersector::Intersections::iterator hitr = intersections.begin();
+    osgUtil::LineSegmentIntersector::Intersections::const_iterator hitr = intersections.cbegin();
 
-    if (!hitr->nodePath.empty() && !(hitr->nodePath.back()->getName().empty())) {
-      mSelectedShape = hitr->nodePath.back()->getName();
-      //std::cout<<"Object identified by name "<<mSelectedShape<<std::endl;
+    if (!hitr->nodePath.empty() && !hitr->nodePath.back()->getName().empty()) {
+      mSelectedVisualizer = hitr->nodePath.back()->getName();
+      //std::cout<<"Object identified by name "<<mSelectedVisualizer<<std::endl;
     } else if (hitr->drawable.valid()) {
-      mSelectedShape = hitr->drawable->className();
-      //std::cout<<"Object identified by its drawable "<<mSelectedShape<<std::endl;
+      mSelectedVisualizer = hitr->drawable->className();
+      //std::cout<<"Object identified by its drawable "<<mSelectedVisualizer<<std::endl;
     }
   }
 }
 
 /*!
- * \brief ViewerWidget::showShapePickContextMenu
+ * \brief ViewerWidget::showVisualizerPickContextMenu
  * \param pos
  */
-void ViewerWidget::showShapePickContextMenu(const QPoint& pos)
+void ViewerWidget::showVisualizerPickContextMenu(const QPoint& pos)
 {
-  QString name = QString::fromStdString(mSelectedShape);
-  //std::cout<<"SHOW CONTEXT "<<name.toStdString()<<" compare "<<QString::compare(name,QString(""))<< std::endl;
-  //the context widget
-  QMenu contextMenu(tr("Context menu"), this);
-  QMenu shapeMenu(name, this);
-  shapeMenu.setIcon(QIcon(":/Resources/icons/animation.svg"));
-  QAction action0(QIcon(":/Resources/icons/undo.svg"), tr("Reset Transparency and Texture"), this);
-  QAction action1(QIcon(":/Resources/icons/transparency.svg"), tr("Change Transparency"), this);
-  QAction action2(QIcon(":/Resources/icons/invisible.svg"), tr("Make Shape Invisible"), this);
-  QAction action3(QIcon(":/Resources/icons/changeColor.svg"), tr("Change Color"), this);
-  QAction action4(QIcon(":/Resources/icons/checkered.svg"), tr("Apply Check Texture"), this);
-  QAction action5(QIcon(":/Resources/icons/texture.svg"), tr("Apply Custom Texture"), this);
-  QAction action6(QIcon(":/Resources/icons/undo.svg"), tr("Remove Texture"), this);
+  AbstractVisualizerObject* visualizer = nullptr;
+  if ((visualizer = mpAnimationWidget->getVisualization()->getBaseData()->getVisualizerObjectByID(mSelectedVisualizer))) {
```

Comment:
I can't the context menu to work. For me the `mSelectedVisualizer` is always empty because `computeIntersections` fails.

Reply:
My bad! This condition should not have been put there. This is fixed in c32e2fc.